pkg/controller/master/upgrade/image_controller.go

```diff
@@ -0,0 +1,47 @@
+package upgrade
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+
+	harvesterv1 "github.com/harvester/harvester/pkg/apis/harvesterhci.io/v1beta1"
+	ctlharvesterv1 "github.com/harvester/harvester/pkg/generated/controllers/harvesterhci.io/v1beta1"
+)
+
+type vmImageHandler struct {
```

Comment:
Can we add some description for each controller so we can easily tell the purpose of it